common/arch/sdl/joy.cpp

```diff
@@ -372,10 +376,19 @@ void joy_init()
 			const auto n_buttons = check_warn_joy_support_limit(SDL_JoystickNumButtons(handle), "button", DXX_MAX_BUTTONS_PER_JOYSTICK);
 			const auto n_hats = check_warn_joy_support_limit(SDL_JoystickNumHats(handle), "hat", DXX_MAX_HATS_PER_JOYSTICK);
 
-			joybutton_text.resize(joybutton_text.size() + n_buttons + (4 * n_hats) + (2 * n_axes));
+			const auto n_virtual_buttons = n_buttons + (4 * n_hats) + (2 * n_axes);
+			joybutton_text.resize(joybutton_text.size() + n_virtual_buttons);
+			joy_key_map.resize(joy_key_map.size() + n_virtual_buttons, 0);
 #if DXX_MAX_BUTTONS_PER_JOYSTICK
 			range_for (auto &&e, enumerate(partial_range(joystick.button_map(), n_buttons), 1))
 			{
+				switch (e.idx) {
+					case 1: joy_key_map[joystick_n_buttons] = KEY_ENTER; break;
+					case 2: joy_key_map[joystick_n_buttons] = KEY_ESC; break;
+					case 3: joy_key_map[joystick_n_buttons] = KEY_SPACEBAR; break;
+					case 4: joy_key_map[joystick_n_buttons] = KEY_DELETE; break;
+					default: break;
+				}
```

Comment:
This switch could be avoided by keeping a `static const` table and pulling the values from there:

```suggestion
    static const unsigned initial_buttons[] = {
        KEY_ENTER,
        KEY_ESC,
        KEY_SPACEBAR,
        KEY_DELETE
    };
    if (e.idx < std::size(initial_buttons)) {
        joy_key_map[joystick_n_buttons] = initial_buttons[e.idx];
    }
```

Reply:
You're right, that would be much better. Shall I change it and make a new PR?